docs/HTTP_SPEC.md

```diff
@@ -14,107 +14,287 @@
+### `X-Request-Id` (request header)
+
+_OPTIONAL_. Used to provide a unique request value that can be correlated with a unique retrieval ID in the logs.
+
+## Request Query Parameters
+
+### `filename` (request query parameter)
+
+Same as [Path Gateway](https://specs.ipfs.tech/http-gateways/path-gateway/#filename-request-query-parameter), but only supports `.car` extensions.
+
+_OPTIONAL_. Used to override the downloaded filename. If provided, it will be included in the `Content-Disposition` header.
+
+If provided, the filename extension is _REQUIRED_ and _MUST_ be `.car`. A missing extension, or any other unspecified extension, will respond with a 400 status code.
+
+Example:
+- `filename=my-file.car` will download a CAR with the name filename `my-file.car`
+
+### `format` (request query parameter)
+
+Same as [Path Gateway](https://specs.ipfs.tech/http-gateways/path-gateway/#format-request-query-parameter), but only supports the `car` format.
+
+_OPTIONAL_. Used to specify the response content type.
+
+This is a URL-friendly alternative to providing an [`Accept`](#accept-request-header) header. _OPTIONAL_ only if an `Accept` header value is provided, otherwise this parameter is _REQUIRED_. Unlike the [`Accept`](#accept-request-header), there is no way to specify media type parameter values. Default media type parameter values will be applied.
+
+If provided, the value _MUST_ be `car`. Any other value will respond with a 400 status code.
+
+Example:
+- `format=car` &rarr; `Accept: application/vnd.ipld.car`
+
+### `dag-scope` (request query parameter)
+
+Specified in [IPIP-402](https://github.com/ipfs/specs/pull/402).
+
+_OPTIONAL_. `dag-scope=(block|entity|all)`. Defaults to `all`.
+
+Describes the shape of the DAG fetched at the terminus of the specified path whose blocks are included in the returned CAR file after the blocks required to traverse path segments. Unspecified values will respond with a 400 status code.
+
+- `block`: Only the root block at the end of the path is returned after blocks required to verify the specified path segments.
+
+- `entity`: Returns only the content at the termination of the `{cid}[/path]` specifier, as well as all blocks from the `cid` to the `path` terminus where a `path` is provided. If the content is found to be UnixFS data, the entire UnixFS entity will be included. i.e. if `{cid}[/path]` terminates at a sharded UnixFS file, or a sharded UnixFS directory, the blocks required to reconsititute the entire file, or directory will be included. If the termination is a UnixFS sharded directory, only the full directory will be included, not the full DAG of the directory's contents.
+
+- `all`: Transmit the entire contiguous DAG that begins at the end of the path query, after blocks required to verify path segments.
+
+When present, the `Etag` response header must include a unique prefix based on the passed scope type.
```

Comment:
```suggestion
When present, the `Etag` response header will include a unique prefix based on the passed scope type.
```

aye, but it doesn't, and we need to fix this, along with the `path` and `dupes` somehow; I think maybe we should just remove this bit entirely until we've address this, it's a bit of a problem as it is now because we're potentially giving the same Etag for different responses.

https://github.com/ipfs/boxo/pull/303/commits/f61f95481041406df46a1781b1daab34b6605650#r1213918777

Reply:
https://github.com/filecoin-project/lassie/pull/285